### PR TITLE
feat: add convert function from 2.0.0 to 2.1.0

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

- add convert function from `2.0.0` to `2.1.0`
- change the logic for convert the spec: in PR we convert the spec from all specs between input and output version, it means that for `1.2.0` to `2.1.0` we convert spec by functions which convert:
  - `1.2.0` -> `2.0.0-rc1`
  - `2.0.0-rc1` -> `2.0.0-rc2`
  - `2.0.0-rc2` -> `2.0.0`
  -  `2.0.0` -> `2.1.0`
  
  By this we don't need logic to write function something like `from2rc1to2_1_0` etc...
  
- added tests

**Related issue(s)**
Resolves https://github.com/asyncapi/converter-js/issues/42

### CODE REVIEW

**Review by jonaslagoni — COMMENTED** [review]
I am wondering, is it not enough that we test with `streetlight`?

Anything extra should be covered by unit tests.

**Comment by magicmatatjahu** [comment]
> I am wondering, is it not enough that we test with streetlight?
>
> Anything extra should be covered by unit tests.

What for example? 😄 We bump only version of spec between `2.0.0` -> `2.1.0`

**Review by derberg — COMMENTED** [review]
as @jonaslagoni is looking into code already, I'm here to just remind you that once we will merge this one, we need to bump it right away in the playground. We also need to discuss if we should offer automated convert from 2.0 to 2.1 as we do with 1->2

File: test/input/2.0.0/gitter-streaming.yml [hunk 1]
```diff
@@ -0,0 +1,137 @@
+asyncapi: 2.0.0
+info:
+  title: Gitter Streaming API
```
Comment (jonaslagoni) [inline test/input/2.0.0/gitter-streaming.yml:3]: Why adding this for `2.0.0` as we already have it present in `1.2.0`. No need to duplicate it I think? 🤔

File: test/input/2.0.0/slack-rtm.yml [hunk 2]
```diff
@@ -0,0 +1,879 @@
+asyncapi: 2.0.0
+info:
```
Comment (jonaslagoni) [inline test/input/2.0.0/slack-rtm.yml:2]: Why adding this for 2.0.0 as we already have it present in 1.2.0. No need to duplicate it I think? 🤔

**Review by jonaslagoni — COMMENTED** [review]
Think it is enough to reuse Streetlight and not the other two, as we already have them as inputs in `1.2.0` 🤔

**Comment by magicmatatjahu** [comment]
@derberg As I wrote in other places, `2.1.0` has only additions so we can convert automatically :)

**Comment by jonaslagoni** [comment]
> we need to bump it right away in the playground

Wouldn't the CD system take care of that? Or is it not enabled for the playground? 🤔

File: test/input/2.0.0/gitter-streaming.yml (hunk 1 above)
Comment (magicmatatjahu) [reply test/input/2.0.0/gitter-streaming.yml]: Removed.

File: test/input/2.0.0/slack-rtm.yml (hunk 2 above)
Comment (magicmatatjahu) [reply test/input/2.0.0/slack-rtm.yml]: Removed.

**Comment by derberg** [comment]
@jonaslagoni bump is not enough. Additional changes are needed to enable or disable 2.0->2.1 conversion

> As I wrote in other places, 2.1.0 has only additions so we can convert automatically :)

@magicmatatjahu thanks for pointing it out, I had no idea. We can convert automatically from 1 to 2 version too. The point is not what we can do technically but what we should do from UX perspective. We need to talk and agree if we do similar modal to playground.asyncapi.io/?url=https://raw.githubusercontent.com/asyncapi/spec/master/examples/1.2.0/slack-rtm.yml or not. Good UX would be to provide such modal, with optional conversion and link to release notes, but this is just my opinion that needs to be validated. Will you drive it forward or should I?

**Review by jonaslagoni — APPROVED** [review]
LGTM 👍

**Comment by magicmatatjahu** [comment]
@derberg I can handle that :)

File: lib/index.js [hunk 3]
```diff
@@ -10,39 +10,63 @@ const {
...
+  if (fromVersion === -1 || toVersion === -1) {
+    console.error(`Cannot convert from ${parsed.asyncapi} to ${version}.`);
+    return;
+  }
+  if (fromVersion > toVersion) {
+    console.error(`Cannot downgrade from ${parsed.asyncapi} to ${version}.`);
+    return;
+  }
+  if (fromVersion === toVersion) {
+    console.error(`Cannot convert to this same version.`);
```
Comment (derberg) [inline lib/index.js:43]:
```suggestion
    console.error(`Cannot convert to the same version.`);
```

**Review by derberg — APPROVED** [review]

**Pull request merged**